packages/dev/core/src/Materials/PBR/openPbrMaterial.ts

```diff
@@ -0,0 +1,2550 @@
+
+        const defines = <OpenPBRMaterialDefines>subMesh.materialDefines;
+        if (this._isReadyForSubMesh(subMesh)) {
+            return true;
+        }
+
+        const scene = this.getScene();
+        const engine = scene.getEngine();
+
+        if (defines._areTexturesDirty) {
+            this._eventInfo.hasRenderTargetTextures = false;
+            this._callbackPluginEventHasRenderTargetTextures(this._eventInfo);
+            this._cacheHasRenderTargetTextures = this._eventInfo.hasRenderTargetTextures;
+            if (scene.texturesEnabled) {
+                // Loop through samplers, check MaterialFlag and whether the texture is ready or not.
+                for (const key in this._samplersList) {
+                    const sampler = this._samplersList[key];
+                    if (sampler.value) {
+                        if (!sampler.value.isReadyOrNotBlocking()) {
+                            return false;
+                        }
+                    }
+                }
+
+                const radianceTexture = this._getRadianceTexture();
+                if (radianceTexture && MaterialFlags.ReflectionTextureEnabled) {
+                    if (!radianceTexture.isReadyOrNotBlocking()) {
+                        return false;
+                    }
+                    if (radianceTexture.irradianceTexture) {
+                        if (!radianceTexture.irradianceTexture.isReadyOrNotBlocking()) {
+                            return false;
+                        }
+                    } else {
+                        // Not ready until spherical are ready too.
+                        if (!radianceTexture.sphericalPolynomial && radianceTexture.getInternalTexture()?._sphericalPolynomialPromise) {
+                            return false;
+                        }
+                    }
+                }
+
+                if (this._environmentBRDFTexture && MaterialFlags.ReflectionTextureEnabled) {
+                    // This is blocking.
+                    if (!this._environmentBRDFTexture.isReady()) {
+                        return false;
+                    }
+                }
+
+                if (OpenPBRMaterial._noiseTexture) {
+                    if (!OpenPBRMaterial._noiseTexture.isReady()) {
+                        return false;
+                    }
+                }
+            }
+        }
+
+        this._eventInfo.isReadyForSubMesh = true;
+        this._eventInfo.defines = defines;
+        this._eventInfo.subMesh = subMesh;
+        this._callbackPluginEventIsReadyForSubMesh(this._eventInfo);
+
+        if (!this._eventInfo.isReadyForSubMesh) {
+            return false;
+        }
+
+        if (defines._areImageProcessingDirty && this._imageProcessingConfiguration) {
+            if (!this._imageProcessingConfiguration.isReady()) {
+                return false;
+            }
+        }
+
+        // Check if Area Lights have LTC texture.
+        if (defines["AREALIGHTUSED"]) {

```

Comment:
We have some new code here related to clustered lights (see **PBRBaseMaterial.ts**).

Reply:
Yup, it's on my list of things to do.